WordPress/src/main/java/org/wordpress/android/ui/posts/PostListFragment.kt

```diff
@@ -235,775 +280,88 @@ class PostListFragment : Fragment(),
 
         // hide the fab so we can animate it
         fabView?.visibility = View.GONE
-        fabView?.setOnClickListener { newPost() }
+        fabView?.setOnClickListener { viewModel.newPost() }
 
         swipeToRefreshHelper = buildSwipeToRefreshHelper(swipeRefreshLayout) {
-            refreshPostList()
+            if (!NetworkUtils.isNetworkAvailable(nonNullActivity)) {
+                swipeRefreshLayout?.isRefreshing = false
+            } else {
+                viewModel.fetchFirstPage()
+            }
         }
-        refreshListManagerFromStore(listDescriptor, shouldRefreshFirstPageAfterLoading = (savedInstanceState == null))
-
         return view
     }
 
     fun handleEditPostResult(resultCode: Int, data: Intent?) {
-        if (resultCode != Activity.RESULT_OK || data == null || !isAdded) {
-            return
-        }
-
-        val localId = data.getIntExtra(EditPostActivity.EXTRA_POST_LOCAL_ID, 0)
-        val post = postStore.getPostByLocalPostId(localId)
-
-        if (post == null) {
-            if (!data.getBooleanExtra(EditPostActivity.EXTRA_IS_DISCARDABLE, false)) {
-                ToastUtils.showToast(nonNullActivity, R.string.post_not_found, ToastUtils.Duration.LONG)
-            }
-            return
-        }
-
-        UploadUtils.handleEditPostResultSnackbars(
-                nonNullActivity,
-                nonNullActivity.findViewById(R.id.coordinator),
-                data,
-                post,
-                site
-        ) {
-            UploadUtils.publishPost(nonNullActivity, post, site, dispatcher)
+        if (resultCode == Activity.RESULT_OK) {
+            viewModel.handleEditPostResult(data)
```

Comment:
I completely agree with the logic, but in this case, it turns out there is no great way to avoid it. Since `UploadUtils.handleEditPostResultSnackbars` requires the data, we could either bypass the `ViewModel` completely or have to refactor the `UploadUtils`. I don't like either option and since this issue is testing related and there are no plans to add tests right now for it, I think it's best to keep it as is and address it when we can. As we discussed on the call, this PR doesn't have to fix everything in post list, it just has to take a step in the right direction.